packages/gestalt/src/Typeahead.js

```diff
@@ -0,0 +1,351 @@
+// @flow strict
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import InputField from './TypeaheadInputField.js';
+import Option from './TypeaheadOption.js';
+import Box from './Box.js';
+import Text from './Text.js';
+import Flyout from './Flyout.js';
+import Layer from './Layer.js';
+
+type TextColors =
+  | 'green'
+  | 'pine'
+  | 'olive'
+  | 'blue'
+  | 'navy'
+  | 'midnight'
+  | 'purple'
+  | 'orchid'
+  | 'eggplant'
+  | 'maroon'
+  | 'watermelon'
+  | 'orange'
+  | 'darkGray'
+  | 'gray'
+  | 'lightGray'
+  | 'red'
+  | 'white';
+
+type OptionObject = {
+  label: string,
+  value: string,
+};
+
+type Props = {|
+  accessibilityLabel: string,
+  id: string,
+  onChange: ({ event: SyntheticInputEvent<>, value: string }) => void,
```

Comment:
@simkessy could we

* Alphabetize these? Looks like `onChange` appears twice in this list.
* Make the events more specific? Same as what we do with `TextField`: `onFocus | ({ event: SyntheticFocusEvent<HTMLInputElement>, value: string }) => void`
![image](https://user-images.githubusercontent.com/127199/85079811-97db1580-b17c-11ea-8f75-3153683b85ae.png)